### DISCUSSION TITLE
Welcome to GMA3Plugins Discussions!

### DISCUSSION THREAD

**Discussion by hossimo** [hossimo/GMA3Plugins#20] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lukas-runge** [comment]
Hey @hossimo,

> I really appreciate the knowledge base for grandMA3 LUA Plugin development that you build up here on GitHub!
> 
> I for myself struggle a lot with adopting LUA in grandMA3 so this really is a huge help as there is not really any documentation on this topic by MAlighting at all. Thus your information is very helpful I often run into challenges I am not able to solve with them. I hope we as a community are able to gather more knowlege on the topic and support each other when one runs into a wall.
> 
> And who knows - maybe MAlighting someday finally values the voluntary work of the community that always goes into developing their missing documentation by trial-and-error. 🙌 

**I am Lukas from Germany 🖐️** 

My starting point for LUA was this video from 2015 by Martijn van Summeren which was actually build purely using macros and image copying. But it inspired me a lot for what was possible even back in 2015: https://youtu.be/EkXAyWWo8z4

Over the years I incorporated LUA more and more into my projects and lighting ideas but never really took off with it in grandMA2. I nearly finished a fully automatic `Color Picker Layout View Builder` for grandMA2 but just before it was done I sadly had no more time and demand for it so I stopped to continue developing the project.

Nowadays - cuz' there is not really live events right now - i started working on a project very similar to the one from Martijn inspiring me back in the day. The goal is to bring a subset of MA3 functions onto an interactive Layout View that is operatable by a person not trained on lighting at all - eg. a facility manager.
Right now I am in the last phase of the project - building the actual thing in the newest version (v1.4.0.2) running into one challenge after another. I am now trying to get in touch with MAlighting for some support, but I don't really expect it to be very helpful - considering the latest answers on LUA I got from them.

Another project I am still working on is porting the [MA2 REAPER Live Marker Sync](https://github.com/locimation/ma2-tc-import/tree/master/TC%20Live)-Plugin from @locimation to MA3: [liveConnect2Reaper](https://github.com/gegenLichtLindenfels/liveConnect2Reaper)

**I am looking forward to connect with other like minded people - sharing knowlege to archive new never thought of solutions! 🚀** 
Kind regards
@lukas-runge

**Reply by hossimo** [reply]
hey @lukas-runge, 

First welcome, and second thanks so much for the donation! It's definitely not required but it's VERY kind and I honestly appreciate it ❤️!

I started this whole github page as a way for me to learn LUA on the MA3 as well as help others with examples. However I ended up putting a ton of time into the Wiki (COVID things 🤷🏿‍♂️) as it was a way for me to remember what I found.

I'm ecstatic that others find my work helpful and I hope we can build a community around LUA on the MA3 and those who love it.

I do think that MA _will_ take a leadership role in how they document and support LUA this time around, and they have some very smart folks who know way more than I do on the official forums, but hopefully we can use this community to help each other!

Thanks again for the kind words, and happy coding!
